frontends/smvparser.y

```diff
@@ -1187,22 +1194,24 @@ next_expr: TOK_NEXT "(" basic_expr ")"{
 
 case_expr: TOK_CASE case_body TOK_ESAC {
   if(enc.module_flat){
-          std::pair<smt::Term,smt::Term> term_last = enc.caseterm_.back();
-          smt::Term cond = term_last.first;
-          smt::Term final_term = term_last.second;
+          std::pair<SMVnode*,SMVnode*> term_last = enc.caseterm_.back();
+          smt::Term cond = term_last.first->getTerm();
+          smt::Term final_term = term_last.second->getTerm();
           enc.caseterm_.pop_back();
           int total_num = enc.caseterm_.size();
+          SMVnode::Type t = SMVnode::Default;
           for (int i = 0; i < total_num; i++){
-            std::pair<smt::Term,smt::Term> term_pair = enc.caseterm_.back();
+            std::pair<SMVnode*,SMVnode*> term_pair = enc.caseterm_.back();
             enc.caseterm_.pop_back();
             case_true = true;
-            smt::Term e = enc.solver_->make_term(smt::Ite, term_pair.first, term_pair.second, final_term);
-            cond = enc.solver_->make_term(smt::BVOr,cond, term_pair.first);
+            t = term_pair.second->getType();
+            smt::Term e = enc.solver_->make_term(smt::Ite, term_pair.first->getTerm(), term_pair.second->getTerm(), final_term);
+            cond = enc.solver_->make_term(smt::BVOr,cond, term_pair.first->getTerm());
```

Comment:
This should be `Or`, not `BVOr`. This might work for boolector, but for all other solvers you can't use a bit-vector operator on booleans. And the conditions should always be boolean.